src/lib/core/rtl/directionality.ts

```diff
@@ -0,0 +1,18 @@
+import {EventEmitter, Injectable} from '@angular/core';
+import {LayoutDirection} from './dir';
+
+/**
+ * A provider to get the direction outside of angular on the `html` or `body` elements
+ */
+@Injectable()
+export class Directionality {
+  value: LayoutDirection;
+  change: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor() {
+    this.value =
+      (document.body.getAttribute('dir') ||
```

Comment:
We still need to account for `dir="auto"`. It looks like `HTMLElemenet.dir` is also `"auto"` when that's set to the attribute, but `getComputedStyle` return either `"ltr"` or `"rtl"`. I'd like to avoid using `getComputedStyle` here, though, since we're already calling it for the theming check. 

Let's just add a `TODO` to handle auto with this background info.